en/docs/insights/choreo-insights-api.md

```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td>
+
+Time filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">metric</td>
+<td valign="top"><a href="#metric">Metric</a>!</td>
+<td>
+
+Metric to retrieve data for
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">groupByFields</td>
+<td valign="top">[<a href="#groupbyfield">GroupByField</a>]!</td>
+<td>
+
+Column/ Field to group the data by
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">groupByValues</td>
+<td valign="top">[<a href="#string">String</a>]!</td>
+<td>
+
+Value to filter by on the groupByField
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">granularity</td>
+<td valign="top"><a href="#string">String</a>!</td>
+<td>
+
+The granularity value for which data is retrieved.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>getCustomReportTopOvertime</strong></td>
+<td valign="top"><a href="#customreporttopovertime">CustomReportTopOvertime</a></td>
+<td>
+
+Returns summary of hits/ latency data for the inquired time frame grouped into timeSpans for the top N values
+for the selected groupByField
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">dataFilter</td>
+<td valign="top"><a href="#datafilter">DataFilter</a>!</td>
+<td>
+
+Data filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">timeFilter</td>
+<td valign="top"><a href="#timefilter">TimeFilter</a>!</td>
+<td>
+
+Time filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">metric</td>
+<td valign="top"><a href="#metric">Metric</a>!</td>
+<td>
+
+Metric to retrieve data for
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">groupByFields</td>
+<td valign="top">[<a href="#groupbyfield">GroupByField</a>]!</td>
+<td>
+
+Column/ Field to group the data by
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">groupByValues</td>
+<td valign="top">[<a href="#string">String</a>]!</td>
+<td>
+
+Value to filter by on the groupByField
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">granularity</td>
+<td valign="top"><a href="#string">String</a>!</td>
+<td>
+
+The granularity value for which data is retrieved.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>getGroupByValues</strong></td>
+<td valign="top">[<a href="#string">String</a>]</td>
+<td>
+
+Returns a list of distinct group by values for the data available for a selected groupByField
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">dataFilter</td>
+<td valign="top"><a href="#datafilter">DataFilter</a>!</td>
+<td>
+
+Data filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">groupByFields</td>
+<td valign="top">[<a href="#groupbyfield">GroupByField</a>]!</td>
+<td>
+
+Column/ Field to get the distinct values for
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">groupByValues</td>
+<td valign="top">[<a href="#string">String</a>]!</td>
+<td>
+
+Value to filter by on the groupByField
```

Comment:
This is not clear. Is the value returned for the groupByField further filtered by something else?

Reply:
`where groupByField == groupByValue`

`groupByValue` is the key value to filter by on the relevant `groupByField`